### DISCUSSION TITLE
How to import users

### DISCUSSION THREAD

**Discussion by hilmanski** [supabase/supabase#1772] [Questions] [answered]
Hi

I want to migrate my project to supabase
I don't see how to migrate my users to supabase

Not database, but adding all previous email I have to "Autentication / User".
so later, my users can just "reset their password" 

[UPDATE]

I see that users are also just a table on our supabase database,
maybe the solution is just import a CSV that match all columns existed

**Comment by silentworks** [comment]
I wouldn't advise on importing directly into the `auth.users` table as there might be other things that need to happen when a user is invited or created in Supabase itself. I did try adding a user in the `auth.users` table and the user did show up in the Supabase UI, but I'm still wary that this might have other side effects I'm not aware of as yet.

I would advise using the supabase library from node with the `service_role` secret and invite users instead. So you can write a node function to loop over an array of users emails and send an invite using the `supabaseAdmin.auth.api.inviteUserByEmail` function.

**Comment by hilmanski** [comment, marked as answer]
Hi @silentworks 

I did try import directly yesterday, but it doesn't work as expected. Can't sign in and show error like "User not found" normally when password is false only show "invalid email or password". So.. something behind the scene don't allow me to do this.



What I end up doing is:
1. Add my user list on new table (temporary table), depend on needs which column to be coming with.
2. Loop all the user list and use singUp supabase auth api. (disable email confirmation, so not bothering other people)
hope it helps for other people.

**Reply by amahmod** [reply]
How did you handle user passwords?

**Reply by tmb5cg** [reply]
@amahmod I have a disclaimer on the login page that says "Existing users: If you're logging in for the first time since MigrationDate, you must reset your password." 

migrated from AWS cognito (amplify auth) to supabase for ref. Not sure if this is the best way to do it

**Reply by RaikkoWeabea** [reply]
Hello everyone! 👋

I'm considering using the `Migrate User` trigger in AWS Cognito to transition users from an existing Cognito user pool to Supabase.

Currently, we have **Cognito user pool A** (with registered users) that we want to migrate. My idea is to create a new **Cognito user pool B** (empty) and use it as an intermediary during authentication.

Here’s the approach:

1. Attach a Migrate User trigger Lambda to **user pool B**.
2. When a user attempts to log in, the Lambda will verify their credentials (email/password) against **user pool A**.
3. If authentication is successful, the Lambda will:
   - **Create the user in Supabase**
   - **Register them in user pool B**, allowing us to track migration progress.
Additionally, we can integrate **third-party auth in Supabase** to accept JWTs from Cognito.

Once all users have migrated to **user pool B**, it means they are also fully migrated to Supabase. At that point, we can remove the Cognito integration and let users authenticate directly in Supabase with the same credentials.

I’m not 100% sure if this approach will work seamlessly, but in theory, it should. It adds an extra step, but this seems like the best way to retain users' email/password credentials.

Would love to hear any thoughts or suggestions! 🚀

**Comment by 108174** [comment]
I created a stored procedure that invites the user using pl-http
